manual/manual/cmds/instrumented-runtime.etex

```diff
@@ -0,0 +1,148 @@
+\chapter{Runtime tracing with the instrumented runtime}
+%HEVEA\cutname{instrumented-runtime.html}
+
+This chapter describes the OCaml instrumented runtime, a runtime variant
+allowing the collection of events and metrics.
+It aims to provide insight into a program's execution.
+
+\section{s:instr-runtime-overview}{Overview}
+
+Once compiled and linked with the instrumented runtime, any OCaml program
+can generate {\em trace files} that can then be read
+and analyzed by users in order to understand specific runtime behaviors.
+
+{\em Common Trace Format} is the general purpose binary
+format used to encode traces.
+A complete trace is comprised of:
+\begin{itemize}
+\item a {\em metadata file}, part of the OCaml distribution
+\item and a {\em trace file}, generated by the runtime\
+  in the program being traced.
+\end{itemize}
+
+For more information on {\em Common Trace Format}, see the website at
+\ifouthtml
+\ahref{https://diamon.org/ctf/}{https://diamon.org/ctf/}.
+\else
+{\tt https://diamon.org/ctf/}
+\fi
+
+\section{s:instr-runtime-enabling}{Enabling runtime instrumentation}
+
+Compiling a program and linking it with the instrumented runtime,
+can be done with {\tt -runtime-variant} flag {\tt i}:
+
+\begin{verbatim}
+       ocamlopt -runtime-variant i program.ml -o program
+\end{verbatim}
```

Comment:
Is there a simple example somewhere of how to configure the runtime variant in a Dune project? (If so, could you link to it as well?)

Reply:
If we use the bytecode compiler, is it possible to choose the runtime at `ocamlrun` invocation time rather than at compilation time? This could make it easier to experiment with the instrumented runtime without having to tweak the build system.

Reply:
Configuring the executable to use the instrumented runtime with dune is as simple as using a `(flag "-runtime-variant=i")` parameter.
As for `ocamlrun`, there is now an `ocamlruni` executable available that is the instrumented runtime.
I can add mentions for both in this section

Reply:
as a side note, for the dune support question, an issue was opened here: https://github.com/ocaml/dune/issues/3500

Reply:
Thanks! For Dune I believe it would be nice to document whatever way currently works. (Most prospective users of the feature are not calling ocamlfind directly, they want to know how to do it through Dune). If the tool gains new feature to make this even easier, we can edit the manual.